tests/unit/moduleapi/rdbloadsave.tcl

```diff
@@ -0,0 +1,143 @@
+set testmodule [file normalize tests/modules/rdbloadsave.so]
+
+start_server {tags {"modules"}} {
+    proc format_command {args} {
+        set cmd "*[llength $args]\r\n"
+        foreach a $args {
+            append cmd "$[string length $a]\r\n$a\r\n"
+        }
+        set _ $cmd
+    }
+
+    r module load $testmodule
+
+    test "Module rdbloadsave sanity" {
+        r test.sanity
+
+        # Try to load non-existing file
+        assert_error {*error*} {r test.rdbload sanity.rdb}
+
+        r set x 1
+        assert_equal OK [r test.rdbsave sanity.rdb]
+
+        r flushdb
+        assert_equal OK [r test.rdbload sanity.rdb]
+        assert_equal 1 [r get x]
+    }
+
+    test "Module rdbloadsave test with pipelining" {
+        r config set save ""
+        r config set loading-process-events-interval-bytes 1024
+        r flushdb
+
+        populate 50000 a 128
+        r set x 111
+        assert_equal [r dbsize] 50001
+
+        assert_equal OK [r test.rdbsave blabla.rdb]
+        r flushdb
+        assert_equal [r dbsize] 0
+
+        # Send commands with pipeline. First command will call RM_RdbLoad() in
+        # the command callback. While loading RDB, Redis can go to networking to
+        # reply -LOADING. By sending commands in pipeline, we verify it doesn't
+        # cause a problem.
+        # e.g. Redis won't try to process next message of the current client
+        # while it is in the command callback for that client   .
+        r write [format_command test.rdbload blabla.rdb]
+        r flush
+        r write [format_command get x]
+        r write [format_command dbsize]
+        r flush
```

Comment:
i wonder why we need `format_command`, we can use a plain `write test.rdbload blabla.rdb` to use the inline protocol (not RESP).
or even just use a deferred client either create another client (`redis_deferring_client`) or set `r deferred 1`.

either way, this doesn't really guarantee redis will read all of these in exactly two event loops.
maybe we should add some delays to the loading (using `key-load-delay` config), and also add a `wait_for_condition` on `[s loading]`?

Reply:
I didn't know about deferring client. I've made the change, please take a look.